opentelemetry/spring-boot-instrumentation/docker-compose.creds.yaml

```diff
@@ -26,8 +28,8 @@ services:
   otelcol:
     # If the collector does not have permission to read the mounted volumes, set
     # USERID=$(id -u) to run the container as the current user
-    user: ${USERID}
+    user: $USERID
     volumes:
-      - ${GOOGLE_APPLICATION_CREDENTIALS:-/dev/null}:/tmp/keys/gcp-credentials.json:ro
+      - ${GOOGLE_APPLICATION_CREDENTIALS?}:${GOOGLE_APPLICATION_CREDENTIALS}:ro
```

Comment:
if I am not mistake this syntax results in failing if the `GOOGLE_APPLICATION_CREDENTIALS` env var is not set.
what is a reason for using this construct in the code sample? see my previous comment about using anti-patterns

Reply:
Correct that is the purpose, to alert the user they need the environment variable with **this** docker compose file. If they want to get credentials from the environment (which doesn't work on a local machine), they can use the other compose file. The two options are explained in [the README](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/2bd2953c40e178965f6f0216650c3e4a7e134877/opentelemetry/spring-boot-instrumentation/README.md#running-the-example) and in the devsite docs.